src/App.js

```diff
@@ -104,62 +98,49 @@ class Board extends React.Component {
 
   //Return position of which square currently in, either moving back, forward, or stay. 
   //Change the possibleNumberBoard and valueBoard
-  positionGenerator(x, y, i, possibleNumberBoard, valueBoard) {
+  positionGenerator(i, possibleNumberBoard, valueBoard) {
 
-    //console.log("x: " + x + ", y: " + y);
-    //console.log("possibleNumberBoard: " + possibleNumberBoard[y * 9 + x] + ", valueBoard: " + valueBoard[y * 9 + x]);
-
-    let position = y * 9 + x;
-
-    console.log("possibleNumberBoard: " + possibleNumberBoard[position]);
-
-    if(possibleNumberBoard[position].length === 0) {
+    if(possibleNumberBoard[i].length === 0) {
 
       //If there are no possble numbers left, refill
-      possibleNumberBoard[position] = [1,2,3,4,5,6,7,8,9];
+      possibleNumberBoard[i] = [1,2,3,4,5,6,7,8,9];
 
-      valueBoard[position] = 0;
-      valueBoard[position - 1] = 0;
-      console.log("Move Back");
+      valueBoard[i] = 0;
+      valueBoard[i - 1] = 0;
 
       //Move backward by 1
       return -2;
     } else {
 
-      //console.log("True");
-
       //Get a random number from array
-      let randomNumberOfArray = possibleNumberBoard[position][Math.floor(Math.random() * possibleNumberBoard[position].length)];
+      let randomNumberOfArray = possibleNumberBoard[i][Math.floor(Math.random() * possibleNumberBoard[i].length)];
       
       //Checks for confliction for row and col
-      let noConflictRow = this.isPossibleRow(randomNumberOfArray, y, valueBoard);
-      let noConflictCol = this.isPossibleCol(randomNumberOfArray, x, valueBoard);
+      let noConflictRow = this.isPossibleRow(randomNumberOfArray, this.returnRow(i), valueBoard);
```

Comment:
There is a performance improvement possible here, but you don't have to make the change in this PR (pull request). If you put these calls directly in the if check, You can save on computation because of something called short-circuit evaluation.

Reply:
So, I call it in the conditions of the if statement?

Reply:
yeah, and you can break it into multiple lines so the line doesn't get too long.

Reply:
Done